test/cea/cea608_memory_unit.js

```diff
@@ -5,238 +5,301 @@
+    };
+
+    const caption = memory.forceEmit(startTime, endTime);
+    expect(caption).toEqual(expectedCaption);
+  });
 
-    // Expect the correct cue to be emitted.
-    const cues = memory.decoder_.getParsedClosedCaptions();
-    expect(cues).toEqual(expectedCaptions);
+  describe('eraseBuffer', () => {
+    it('erases the entire buffer', () => {
+      const startTime = 1;
+      const endTime = 2;
+      const text = '0123456789abcde';
+
+      // Add characters to the memory buffer.
+      for (const c of text) {
+        memory.addChar(CharSet.BASIC_NORTH_AMERICAN,
+            c.charCodeAt(0));
+        memory.setRow(memory.getRow() + 1); // increment row
+      }
 
-    // Erase the memory and currently buffered cues.
-    memory.eraseBuffer();
-    memory.decoder_.clearParsedClosedCaptions();
+      // Erase the entire memory buffer.
+      memory.eraseBuffer();
 
-    // Force out the new memory.
-    memory.forceEmit(t1, t2);
+      // Force out the memory buffer.
+      const caption = memory.forceEmit(startTime, endTime);
 
-    // Expect the forced out memory to be blank. We just cleared it.
-    expect(memory.decoder_.getParsedClosedCaptions()).toEqual([]);
+      // Expect the forced out memory to be blank. We just cleared it.
+      expect(caption).toBe(null);
+    });
   });
 
-  it('erases the entire buffer', () => {
-    const t1 = 1;
-    const t2 = 2;
-    const text = 'test';
-    const expectedText1 = 't\ne\ns\nt';
-    const expectedText2 = 's\nt\nt\ne';
-    for (const c of text) {
-      memory.addChar(shaka.cea.Cea608Memory.CharSet.BASIC_NORTH_AMERICAN,
-          c.charCodeAt(0));
-      memory.setRow(memory.getRow() + 1); // increment row
-    }
-    const expectedCaptions1 = [createDefaultClosedCaption(t1, t2, expectedText1)];
-    memory.forceEmit(t1, t2);
-
-    // Expect the correct cue to be emitted.
-    const cues = memory.decoder_.getParsedClosedCaptions();
-    expect(cues).toEqual(expectedCaptions1);
-
-    // Move + clear the first 2 rows, and clear currently buffered cues.
-    const srcRowIdx = 1;
-    const dstRowIdx = 5;
-    const rowsToMove = 2;
-    memory.moveRows(dstRowIdx, srcRowIdx, rowsToMove);
-    memory.resetRows(srcRowIdx, rowsToMove - 1);
-    memory.decoder_.clearParsedClosedCaptions();
-
-    // Force out the new memory.
-    memory.forceEmit(t1, t2);
-
-    const expectedCaptions2 = [createDefaultClosedCaption(t1, t2, expectedText2)];
-    expect(memory.decoder_.getParsedClosedCaptions()).toEqual(expectedCaptions2);
+  describe('moveRows', () => {
+    it('moves a set number of rows to a new position in the buffer', () => {
+      const startTime = 1;
+      const endTime = 2;
+      const text = 'test';
+
+      // Add the text to the buffer, each character on separate rows.
+      for (const c of text) {
+        memory.addChar(CharSet.BASIC_NORTH_AMERICAN,
+            c.charCodeAt(0));
+        memory.setRow(memory.getRow() + 1); // increment row
+      }
```

Comment:
I had to work this out on paper to follow along.  Some diagrams in comments would help the reader.   For example:

```js
// At this point, the rows looks like:
// [1]: t
// [2]: e
// [3]: s
// [4]: t
```

After move, a diagram like:

```js
// [1]:
// [2]:
// [3]: s
// [4]: t
// [5]: t
// [6]: e
```

Those comments would have saved me the trouble of working it out on paper.